### PR TITLE
Styled CurrencyField

### PR DESCRIPTION
**Pull request by dawinValenzuela**
## Please delete checklist items and sections that are not relevant

**Checklist**

- [x] New feature (non-breaking change which adds functionality)

**Please check if the PR fulfills these requirements**

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Storybook or docs have been added / updated (for bug fixes / features)

**What is the ticket / issue associated with this change?** (Link to the ticket or issue)
https://trello.com/c/95Nr1LgN/94-migrate-composer-forms-to-styled-components

### CODE REVIEW

File: packages/styled-components/src/components/primitives/Form/CurrencyField/CurrencyField.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      required,
+      inputMode,
+      testId = 'currencyField',
+      customCss,
+      onChange,
+    }: Props,
+    ref
+  ) => (
+    <StyledWrapper>
+      <StyledIconWrapper className={cn(shape)}>
```
Comment (guilean) [inline packages/styled-components/src/components/primitives/Form/CurrencyField/CurrencyField.tsx:34]: Remove cn

**Review by guilean — CHANGES_REQUESTED** [review]

File: packages/styled-components/src/components/primitives/Form/CurrencyField/CurrencyField.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+export default {
+  title: 'Primitives/Form/CurrencyField',
+  component: CurrencyField,
+};
+
+export const Shapes = () => (
+  <>
+    <h1>CurrencyField shapes</h1>
+    {Object.keys(CurrencyField.Shapes).map((shape: any) => (
+      <>
```
[the remark this answers is not in the document]
Comment (dawinValenzuela) [reply packages/styled-components/src/components/primitives/Form/CurrencyField/CurrencyField.stories.tsx]: done!

File: packages/styled-components/src/components/primitives/Form/CurrencyField/CurrencyField.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      </>
+    ))}
+  </>
+);
+
+export const Modes = () => (
+  <>
+    <h1>CurrencyField Modes</h1>
+    {Object.keys(CurrencyField.Modes).map((mode: any) => (
+      <>
```
[the remark this answers is not in the document]
Comment (dawinValenzuela) [reply packages/styled-components/src/components/primitives/Form/CurrencyField/CurrencyField.stories.tsx]: done!

File: packages/styled-components/src/components/primitives/Form/CurrencyField/CurrencyField.tsx (hunk 1 above)
Comment (dawinValenzuela) [reply packages/styled-components/src/components/primitives/Form/CurrencyField/CurrencyField.tsx]: done!

**Review by tkanzakic — APPROVED** [review]

File: packages/styled-components/src/components/primitives/Form/CurrencyField/CurrencyField.test.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
+import React from 'react';
```
Comment (guilean) [inline packages/styled-components/src/components/primitives/Form/CurrencyField/CurrencyField.test.tsx:1]: Could we rewrite these tests to look like `ExpandableTextfield` tests? We have a lot of repeated code

Comment (dawinValenzuela) [reply packages/styled-components/src/components/primitives/Form/CurrencyField/CurrencyField.test.tsx:1]: done!

File: packages/styled-components/src/components/primitives/Form/CurrencyField/CurrencyField.test.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    expect(input).toBeDisabled();
+  });
+  it('should render inputMode', () => {
+    givenComponentRendered({ inputMode: CurrencyField.Modes.Numeric });
+    const input = screen.getByTestId(testId);
+    expect(input).toHaveAttribute('inputmode', 'numeric');
+  });
+  it('should call onChange', () => {
+    const mockOnChange = jest.fn();
+    givenComponentRendered({ required: true, onChange: mockOnChange });
```
Comment (guilean) [inline packages/styled-components/src/components/primitives/Form/CurrencyField/CurrencyField.test.tsx:45]: required unneded

Comment (dawinValenzuela) [reply packages/styled-components/src/components/primitives/Form/CurrencyField/CurrencyField.test.tsx:45]: done!

**Pull request merged**